services/api/src/api/routes/filter.py

```diff
@@ -0,0 +1,111 @@
+                    offset = int(request.query_params.get("offset", 0))
+                    if offset < 0:
+                        raise InvalidParameterError(message="Offset must be positive")
+                    length = int(request.query_params.get("length", MAX_ROWS))
+                    if length < 0:
+                        raise InvalidParameterError("Length must be positive")
+                    if length > MAX_ROWS:
+                        raise InvalidParameterError(f"Length must be less than or equal to {MAX_ROWS}")
+                    logger.info(
+                        f'/filter, dataset={dataset}, config={config}, split={split}, where="{where}",'
+                        f" offset={offset}, length={length}"
+                    )
+                with StepProfiler(method="filter_endpoint", step="check authentication"):
+                    # If auth_check fails, it will raise an exception that will be caught below
+                    auth_check(
+                        dataset=dataset,
+                        external_auth_url=external_auth_url,
+                        request=request,
+                        hf_jwt_public_key=hf_jwt_public_key,
+                        hf_jwt_algorithm=hf_jwt_algorithm,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                    )
+                with StepProfiler(method="filter_endpoint", step="get config-parquet-metadata from cache"):
+                    parquet_file_metadata_items, revision = get_config_parquet_metadata_from_cache(
+                        dataset=dataset, config=config, processing_graph=processing_graph
+                    )
+                with StepProfiler(method="filter_endpoint", step="create response"):
+                    response = {"status": "ok"}
+                with StepProfiler(method="filter_endpoint", step="generate the OK response"):
+                    return get_json_ok_response(content=response, max_age=max_age_long, revision=revision)
+            except Exception as e:
+                error = e if isinstance(e, ApiCustomError) else UnexpectedError("Unexpected error.", e)
+                with StepProfiler(method="filter_endpoint", step="generate API error response"):
+                    return get_json_api_error_response(error=error, max_age=max_age_short, revision=revision)
+
+    return filter_endpoint
+
+
+def get_config_parquet_metadata_from_cache(
+    dataset: str, config: str, processing_graph: ProcessingGraph
+) -> tuple[list[ParquetFileMetadataItem], Optional[str]]:
+    config_parquet_metadata_processing_steps = processing_graph.get_config_parquet_metadata_processing_steps()
+    if not config_parquet_metadata_processing_steps:
+        raise RuntimeError("No processing steps are configured to provide config-parquet-metadata response.")
+    cache_kinds = [step.cache_kind for step in config_parquet_metadata_processing_steps]
+    try:
+        result = get_previous_step_or_raise(
```

Comment:
The function `get_previous_step_or_raise` is not tested anywhere.

Reply:
could you add one, or open an issue?

Reply:
Done:
- #1908